crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/generated_acir.rs

```diff
@@ -95,57 +95,97 @@ impl GeneratedAcir {
 }
 
 impl GeneratedAcir {
-    /// Computes lhs mod 2^rhs
+    /// Computes lhs = 2^{rhs_bit_size} * q + r
     ///
-    /// `max_bits` is the upper-bound on the bit_size of the object that `lhs` is representing.
-
-    /// An example; max_bits would be 32, if lhs was representing a u32 at a higher level.
+    /// For example, if we had a u32:
+    ///     - `rhs` would be `32`
+    ///     - `max_bits` would be the size of `lhs`
+    ///
+    /// Take the following code:
+    /// ``
+    ///   fn main(x : u32) -> u32 {
+    ///     let a = x + x; (L1)
+    ///     let b = a * a; (L2)
+    ///     b + b (L3)
+    ///   }
+    /// ``
+    ///
+    ///  Call truncate only on L1:
+    ///     - `rhs` would be `32`
+    ///     - `max_bits` would be `33` due to the addition of two u32s
+    ///  Call truncate only on L2:
+    ///     - `rhs` would be `32`
+    ///     - `max_bits` would be `66` due to the multiplication of two u33s `a`
+    ///  Call truncate only on L3:
+    ///     -  `rhs` would be `32`
+    ///     - `max_bits` would be `67` due to the addition of two u66s `b`
```

Comment:
On L2, why would max_bits be 66 instead of 64 if the value was just truncated to 32 bits previously? By "call truncate only on l2" are we referring to "if truncate was included only after l2" or "this comment refers to only the truncate after l2"?

I'm trying to follow the logic for why we need `max_bits` at all still compared to having a simpler `Truncate { num_bits_to_truncate_to }` instruction. It seems like `max_bits` is meant for tracking metadata but it is still used for a range_constraint and a push_addition_term below.